swarm/network/stream/stream.go

```diff
@@ -938,16 +938,22 @@ It returns a map of node IDs with an array of string representations of Stream o
 func (api *API) GetPeerSubscriptions() map[string][]string {
 	//create the empty map
 	pstreams := make(map[string][]string)
+
 	//iterate all streamer peers
+	api.streamer.peersMu.RLock()
```

Comment:
I don't really like that we are accessing this mutex here. What do you thin about getting peers from the registry?
 
My two reasons:
1) This mutex so far was only accessed  by `Registry` methods.  To me it looks like we are introducing unnecessary coupling between two structs, i.e., `Registry` and `API`. Thus we break the [Law_of_Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) by talking to *strangers*. 
2) This is just something I'm thinking about... the loop we iterate is n^2 . This is an API and usually we should  try to minimize time spent in protected areas. Do we potentially block other API calls?

![image](https://user-images.githubusercontent.com/1735667/52631718-d2b1a700-2ebf-11e9-99ea-17b6a5664518.png)

Reply:
This PR is limited to fixing the race conditions. I think that another issue should be created if the  implementation of GetPeerSubscriptions should be refactored. I think that we would need more changes to make it better, not just functional.

Reply:
I was thinking about that scope limit, too. However these improvements should have really low cost in a healthy code base. If you think the suggested method extraction is too risky because of insufficient test coverage I'm fine with your decision.

Reply:
It would be good to address the code improvements in a different PR to isolate it if any problems arise. It would be good to have an issue recorded on this. Would you mind creating one with some guidance  about improvements?

Reply:
After looking more at this code, I decided this probably does not worth the investment. `API` is relatively small and we don't leave the `stream.go` file. Also this code is relatively new, so improving it should not belong to this PR for sure.

Reply:
Cool, thanks.